contracts/Tokens/Prime/README.md

```diff
@@ -0,0 +1,345 @@
+
+This is how we will calculate the user rewards:
+
+```jsx
+rewards = (rewardIndex - userRewardIndex) * scoreOfUser
+```
+
+Then we will update the `userRewardIndex` (`interests[market][account]`) to their current global values.
+
+## Income collection and distribution
+
+Interest reserves (part of the protocol income) from core pool markets are sent to the PSR ([Protocol Share Reserve](https://github.com/VenusProtocol/protocol-reserve/blob/develop/contracts/ProtocolReserve/ProtocolShareReserve.sol)) contract. Based on configuration a certain percentage of spread income from Prime markets is reserved for prime token holders. The interest reserves will be sent to the PSR every 10 blocks (this can be changed by the community via [VIP](https://app.venus.io/governance)).
+
+The PSR has a function `releaseFunds` that needs to be invoked to release the funds to the Prime contract, and it also has a function `getUnreleasedFunds` to get the unreleased funds for a given destination target.
+
+Prime contract takes the total released + unreleased funds and distributes to prime token holders each block. The distribution is proportional to the score of the prime token holders.
+
+When a user claims their rewards and if the contract doesn’t have enough funds then we trigger the release of funds from PSR to Prime contract in the same transaction i.e., in the `claim` function.
+
+Two key integration points in the PSR contract related with the Prime contract:
+
+* When a distribution configuration changes the PSR contract first calls `accrueInterest` on the Prime contract and releases funds to Prime (also invoking `updateAssetsState` function on the Prime contract) before saving the changes. This ensures pending funds for Prime are distributed and allocated per the schema by which they were accrued before applying new changes to the percentage of the protocol income allocated to Prime.
+
+* Prior to releasing funds to the Prime contract, the PSR contract calls `accrueInterest` and then after funds are sent it calls `updateAssetsState`.
+
+More information about Income collection and distribution [here](https://docs-v4.venus.io/whats-new/automatic-income-allocation).
+
+## Update cap multipliers and alpha
+
+Market multipliers and alpha can be updated at anytime and need to be propagated to all users. Changes will be gradually applied to users as they borrow/supply assets and their individual scores are recalculated. This strategy has limitations because the scores will be wrong in aggregate.
+
+To mitigate this issue, Venus will supply a script that will use the permission-less function `updateScores` to update the scores of all users. This script won’t pause the market or prime contract. This script will update scores in multiple transactions because it will run out of gas trying to update all scores in 1 transaction. 
```

Comment:
```suggestion
To mitigate this issue, Venus will supply a script that will use the permission-less function `updateScores` to update the scores of all users. This script won’t pause the market or prime contract.  Scores need to be updated in multiple transactions because it will run out of gas trying to update all scores in 1 transaction. 
```